src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,126 @@
+package mate.academy.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String CREATE_SQL_OPERATION =
+            "INSERT INTO books (title, price) VALUES (?, ?)";
+    private static final String GET_SQL_OPERATION =
+            "SELECT * FROM books WHERE id = (?)";
+    private static final String UPDATE_SQL_OPERATION =
+            "UPDATE books SET title = (?), price = (?) WHERE id = (?)";
+    private static final String DELETE_SQL_OPERATION = "DELETE FROM books WHERE id = (?)";
+    private static final String GET_ALL_SQL_OPERATION = "SELECT * FROM books";
+
+    @Override
+    public Book create(Book book) {
+        if (book == null) {
+            throw new DataProcessingException("Error: This book is null!");
+        }
+        try (Connection connection = ConnectionUtil.connect();
+                PreparedStatement statement
+                        = connection.prepareStatement(CREATE_SQL_OPERATION,
+                             Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Expected to insert at least one row,"
+                        + " but inserted 0 rows");
+            }

```

Comment:
Duplication of code.

Reply:
the messages are different

Reply:
I believe you can adapt the message like this: Expected to affect at least one row but 0 rows affected" and move logic to separate method